test/e2e/tests/test_firewall.py

```diff
@@ -52,136 +48,71 @@ def simple_firewall(request, simple_firewall_policy):
     replacements["TRAFFIC_TYPE"] = "ALL"
     replacements["FIREWALL_POLICY_ARN"] = policy_arn
 
-    # Load Network Firewall CR
-    resource_data = load_networkfirewall_resource(
-        "firewall",
-        additional_replacements=replacements,
-    )
-
-    # Create k8s resource
-    ref = k8s.CustomResourceReference(
-        CRD_GROUP,
-        CRD_VERSION,
-        RESOURCE_PLURAL,
+    ref, cr = create_firewall_resource(
+        "firewalls",
         fw_resource_name,
-        namespace="default",
+        "firewall",
+        replacements,
     )
-    k8s.create_custom_resource(ref, resource_data)
-
-    time.sleep(CREATE_WAIT_AFTER_SECONDS)
-
-    # Get latest Network Firewall CR
-    cr = k8s.wait_resource_consumed_by_controller(ref)
 
-    assert cr is not None
-    assert k8s.get_resource_exists(ref)
-
-    yield (ref, cr)
-
-    # Try to delete, if doesn't already exist
-    try:
-        _, deleted = k8s.delete_custom_resource(ref, 3, 10)
-        assert deleted
-    except:
-        pass
+    assert cr["status"]["firewall"]["firewallPolicyARN"] == policy_arn
 
-@pytest.fixture
```

Comment:
I needed to use this function twice in `simple_firewall`...so had to stop using this as a fixture